respx/models.py

```diff
@@ -44,30 +44,36 @@ def clone_response(response: httpx.Response, request: httpx.Request) -> httpx.Re
 
 class Call(NamedTuple):
     request: httpx.Request
-    response: Optional[httpx.Response]
+    optional_response: Optional[httpx.Response]
+
+    @property
+    def response(self) -> httpx.Response:
+        if self.optional_response is None:
+            raise ValueError(f"{self!r} has no response")
+        return self.optional_response
 
 
 class CallList(list, mock.NonCallableMock):
-    def __init__(self, *args, name="respx", **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, *args: Sequence[Call], name: Any = "respx") -> None:
```

Comment:
Perhaps `name: str | None` to mimic `NonCallableMock`?

```suggestion
    def __init__(self, *args: Sequence[Call], name: str | None = "respx") -> None:
```

Ref: https://github.com/python/typeshed/blob/c9346f32e10fc631e691745e2a8c24bc3216642f/stdlib/unittest/mock.pyi#L110

Reply:
Totally agree, but it's not possible due to this "unfortunate" solution ... https://github.com/lundberg/respx/blob/master/respx/models.py#L115

I use the route instance as mock `name` to allow it to be dynamic/lazy.

To give some context, the name is only visible on assertion errors, e.g. when accessing NonCallableMock `.assert_called_once` etc, and passing the route as _name_ is to render an up-to-date repr/str when that happens.